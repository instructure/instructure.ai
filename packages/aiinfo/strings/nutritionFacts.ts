// biome-ignore-all assist/source/useSortedKeys: Required order for Strings

import type { Strings } from "../types";

const strings: Strings = {
	en: {
		modalLabel: "This is a modal for AI facts",
		title: "Nutrition Facts",
		data: [
			{
				blockTitle: "Model & Data",
				segmentData: [
					{
						segmentTitle: "Base Model",
						description:
							"The foundational AI on which further training and customizations are built.",
					},
					{
						segmentTitle: "Trained with User Data",
						description:
							"Indicates the AI model has been given customer data in order to improve its results.",
					},
					{
						segmentTitle: "Data Shared with Model",
						description:
							"Indicates which training or operational content was given to the model.",
					},
				],
			},
			{
				blockTitle: "Privacy & Compliance",
				segmentData: [
					{
						segmentTitle: "Data Retention",
						description: "How long the model stores customer data.",
					},
					{
						segmentTitle: "Data Logging",
						description:
							"Recording the AI's performance for auditing, analysis, and improvement.",
					},
					{
						segmentTitle: "Regions Supported",
						description:
							"The locations where the AI model is officially available and supported.",
					},
					{
						segmentTitle: "PII",
						description:
							"Sensitive data that can be used to identify an individual.",
					},
				],
			},
			{
				blockTitle: "Outputs",
				segmentData: [
					{
						segmentTitle: "AI Settings Control",
						description:
							"The ability to turn the AI on or off within the product.",
					},
					{
						segmentTitle: "Human in the Loop",
						description:
							"Indicates if a human is involved in the AI's process or output.",
					},
					{
						segmentTitle: "Guardrails",
						description:
							"Preventative safety mechanisms or limitations built into the AI model.",
					},
					{
						segmentTitle: "Expected Risks",
						description: "Any risks the model may pose to the user.",
					},
					{
						segmentTitle: "Intended Outcomes",
						description:
							"The specific results the AI model is meant to achieve.",
					},
				],
			},
		],
		closeButtonText: "Close",
		closeIconButtonScreenReaderLabel: "Close",
		triggerText: "Nutrition Facts",
	},
};

export { strings };
