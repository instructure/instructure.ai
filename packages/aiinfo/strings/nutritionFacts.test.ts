import { describe, expect, it } from "vitest";
import { strings as nutritionFactsStrings } from "./nutritionFacts";

describe("nutritionFacts strings", () => {
	it("exposes 'en' locale", () => {
		expect(nutritionFactsStrings).toHaveProperty("en");
	});

	it("has required top-level keys with correct literals", () => {
		const nf = nutritionFactsStrings.en as Record<string, unknown>;
		const required = [
			"modalLabel",
			"title",
			"data",
			"closeButtonText",
			"closeIconButtonScreenReaderLabel",
			"triggerText",
		];
		required.forEach((k) => {
			expect(nf[k]).not.toBeUndefined();
		});
		expect(nf.title).toBe("AI Nutrition Facts");
		expect(nf.triggerText).toBe("Nutrition Facts");
		expect(nf.closeButtonText).toBe("Close");
		expect(nf.closeIconButtonScreenReaderLabel).toBe("Close");
		expect(nf.closeButtonText).toBe(nf.closeIconButtonScreenReaderLabel);
		expect(nf.modalLabel).toBe("This is a modal for AI facts");
	});

	it("has three data blocks with expected lengths and titles", () => {
		const nf = nutritionFactsStrings.en;
		expect(Array.isArray(nf.data)).toBe(true);
		expect(nf.data).toHaveLength(3);

		const [modelBlock, privacyBlock, outputsBlock] = nf.data;

		expect(modelBlock.blockTitle).toBe("Model & Data");
		expect(modelBlock.segmentData).toHaveLength(3);
		expect(modelBlock.segmentData.map((s) => s.segmentTitle)).toEqual([
			"Base Model",
			"Trained with User Data",
			"Data Shared with Model",
		]);

		expect(privacyBlock.blockTitle).toBe("Privacy & Compliance");
		expect(privacyBlock.segmentData).toHaveLength(4);
		expect(privacyBlock.segmentData.map((s) => s.segmentTitle)).toEqual([
			"Data Retention",
			"Data Logging",
			"Regions Supported",
			"PII",
		]);

		expect(outputsBlock.blockTitle).toBe("Outputs");
		expect(outputsBlock.segmentData).toHaveLength(5);
		expect(outputsBlock.segmentData.map((s) => s.segmentTitle)).toEqual([
			"AI Settings Control",
			"Human in the Loop",
			"Guardrails",
			"Expected Risks",
			"Intended Outcomes",
		]);
	});

	it("has correct total number of segments", () => {
		const nf = nutritionFactsStrings.en;
		const allSegments = nf.data.flatMap((b) => b.segmentData);
		expect(allSegments).toHaveLength(12);
	});

	it("ensures all segment titles are unique and capitalized", () => {
		const nf = nutritionFactsStrings.en;
		const titles = nf.data.flatMap((b) =>
			b.segmentData.map((s) => s.segmentTitle),
		);
		const set = new Set(titles);
		expect(set.size).toBe(titles.length);
		titles.forEach((t) => {
			expect(/^[A-Z]/.test(t)).toBe(true);
		});
	});

	it("ensures all segment descriptions are present and non-empty", () => {
		const nf = nutritionFactsStrings.en;
		nf.data.forEach((block) => {
			block.segmentData.forEach((seg) => {
				expect(typeof seg.description).toBe("string");
				expect(seg.description.length).toBeGreaterThan(0);
			});
		});
	});

	it("first segment title is 'Base Model' (regression guard used elsewhere)", () => {
		const nf = nutritionFactsStrings.en;
		expect(nf.data[0].segmentData[0].segmentTitle).toBe("Base Model");
	});

	it("no undefined at top level", () => {
		const nf = nutritionFactsStrings.en as Record<string, unknown>;
		Object.entries(nf).forEach(([, v]) => {
			expect(v).not.toBeUndefined();
		});
	});

	it("blocks have only expected keys", () => {
		const nf = nutritionFactsStrings.en;
		nf.data.forEach((block) => {
			expect(Object.keys(block).sort()).toEqual(["blockTitle", "segmentData"]);
		});
	});

	it("segment entries have only expected keys", () => {
		const nf = nutritionFactsStrings.en;
		nf.data.forEach((block) => {
			block.segmentData.forEach((seg) => {
				expect(Object.keys(seg).sort()).toEqual([
					"description",
					"segmentTitle",
				]);
			});
		});
	});

	it("snapshot of english nutrition facts strings", () => {
		expect(nutritionFactsStrings.en).toMatchSnapshot();
	});
});
