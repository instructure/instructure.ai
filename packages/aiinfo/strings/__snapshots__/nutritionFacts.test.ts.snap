// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nutritionFacts strings > snapshot of english nutrition facts strings 1`] = `
{
  "closeButtonText": "Close",
  "closeIconButtonScreenReaderLabel": "Close",
  "data": [
    {
      "blockTitle": "Model & Data",
      "segmentData": [
        {
          "description": "The foundational AI on which further training and customizations are built.",
          "segmentTitle": "Base Model",
        },
        {
          "description": "Indicates the AI model has been given customer data in order to improve its results.",
          "segmentTitle": "Trained with User Data",
        },
        {
          "description": "Indicates which training or operational content was given to the model.",
          "segmentTitle": "Data Shared with Model",
        },
      ],
    },
    {
      "blockTitle": "Privacy & Compliance",
      "segmentData": [
        {
          "description": "How long the model stores customer data.",
          "segmentTitle": "Data Retention",
        },
        {
          "description": "Recording the AI's performance for auditing, analysis, and improvement.",
          "segmentTitle": "Data Logging",
        },
        {
          "description": "The locations where the AI model is officially available and supported.",
          "segmentTitle": "Regions Supported",
        },
        {
          "description": "Sensitive data that can be used to identify an individual.",
          "segmentTitle": "PII",
        },
      ],
    },
    {
      "blockTitle": "Outputs",
      "segmentData": [
        {
          "description": "The ability to turn the AI on or off within the product.",
          "segmentTitle": "AI Settings Control",
        },
        {
          "description": "Indicates if a human is involved in the AI's process or output.",
          "segmentTitle": "Human in the Loop",
        },
        {
          "description": "Preventative safety mechanisms or limitations built into the AI model.",
          "segmentTitle": "Guardrails",
        },
        {
          "description": "Any risks the model may pose to the user.",
          "segmentTitle": "Expected Risks",
        },
        {
          "description": "The specific results the AI model is meant to achieve.",
          "segmentTitle": "Intended Outcomes",
        },
      ],
    },
  ],
  "modalLabel": "This is a modal for AI facts",
  "title": "AI Nutrition Facts",
  "triggerText": "Nutrition Facts",
}
`;
